### PR TITLE
add datatbase, dao's and entities

### PR DESCRIPTION
**Pull request by zhenyabigel**
add room datatbase, dao's and entities

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -52,12 +51,39 @@ android {
...
+    implementation("androidx.room:room-rxjava2:$room_version")
+
+    // optional - RxJava3 support for Room
+    implementation("androidx.room:room-rxjava3:$room_version")
+
+    // optional - Guava support for Room, including Optional and ListenableFuture
+    implementation("androidx.room:room-guava:$room_version")
+
+    // optional - Test helpers
+    testImplementation("androidx.room:room-testing:$room_version")
```
Comment (Razor0744) [inline app/build.gradle.kts]: remove comments and unused implementation

File: app/src/main/java/by/zhenyabigel/bankingapplication/data/dao/AccountDao.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    fun getListAsFlow(): Flow<List<AccountEntity>>
+    @Query("SELECT * FROM ${AccountEntity.TABLE_NAME} WHERE id=:id ")
+    fun getItem(id: UUID): AccountEntity
+    @Query("SELECT * FROM ${AccountEntity.TABLE_NAME} WHERE id=:id ")
+    fun getItemAsFlow(id: UUID): Flow<AccountEntity>
+    @Upsert // INSERT OR REPLACE в БД
+    suspend fun save(e: AccountEntity)
+    @Query("DELETE FROM ${AccountEntity.TABLE_NAME} WHERE id = :id")
+    suspend fun delete(id: UUID)
+}
```
Comment (Razor0744) [inline app/src/main/java/by/zhenyabigel/bankingapplication/data/dao/AccountDao.kt]: pls don't forget about spaces

File: app/src/main/java/by/zhenyabigel/bankingapplication/data/dao/TransactionDao.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    @Query("SELECT * FROM ${TransactionEntity.TABLE_NAME} WHERE id=:id ")
+    fun getItemAsFlow(id: UUID): Flow<TransactionEntity>
+    @Query("SELECT * FROM ${TransactionEntity.TABLE_NAME} WHERE id=:id ")
+    fun getItem(id: UUID): TransactionEntity
+    @Upsert
+    suspend fun save(e: TransactionEntity)
+
+    @Query("DELETE FROM ${TransactionEntity.TABLE_NAME} WHERE id = :id")
+    suspend fun delete(id: UUID)
+}
```
Comment (Razor0744) [inline app/src/main/java/by/zhenyabigel/bankingapplication/data/dao/TransactionDao.kt]: the same

File: app/src/main/java/by/zhenyabigel/bankingapplication/data/datasource/AccountDataSource.kt [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    fun getAccounts(): List<AccountEntity> = accountDao.getList()
+    fun getAccountAsFlow(id: UUID): Flow<AccountEntity> = accountDao.getItemAsFlow(id)
+    fun getAccount(id: UUID): AccountEntity = accountDao.getItem(id)
+    fun getAccountsList(): List<AccountEntity> = accountDao.getList()
+    suspend fun upsert(account: AccountEntity) {
+        accountDao.save(account)
+    }
+    suspend fun delete(id: UUID) {
+        accountDao.delete(id)
+    }
```
Comment (Razor0744) [inline app/src/main/java/by/zhenyabigel/bankingapplication/data/datasource/AccountDataSource.kt]: here also

File: app/src/main/java/by/zhenyabigel/bankingapplication/data/entities/TransactionEntity.kt [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    fun toTransaction(): Transaction {
+        return Transaction(
+            id = id,
+            company = company,
+            status=status,
+            transactionNumber = transactionNumber,
+            date = date,
+            amount = amount.toString()
+        )
+    }
```
Comment (Razor0744) [inline app/src/main/java/by/zhenyabigel/bankingapplication/data/entities/TransactionEntity.kt]: pls tell me what is that?

File: app/src/main/java/by/zhenyabigel/bankingapplication/data/entities/TransactionEntity.kt [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    @PrimaryKey val id: UUID,
+    @ColumnInfo(name = "company") val company: String,
+    @ColumnInfo(name = "transaction_number") val transactionNumber: String,
+    @ColumnInfo(name = "date") val date: String,
+    @ColumnInfo(name = "status") val status: String,
+    @ColumnInfo(name = "amount") val amount: Int
+) {
+    internal companion object {
+        const val TABLE_NAME = "transaction_table" // <- удобно потом ссылаться будет
+    }
```
Comment (Razor0744) [inline app/src/main/java/by/zhenyabigel/bankingapplication/data/entities/TransactionEntity.kt]: don't use const like this

File: app/src/main/java/by/zhenyabigel/bankingapplication/data/model/Account.kt [hunk 7]
```diff
@@ -10,4 +11,11 @@ data class Account(
...
     @DrawableRes
     val cover: Int,
+)
+internal fun Account.toEntity() = AccountEntity(
+    id = id,
+    number = number,
+    walletID = walletID,
+    ownerName = ownerName,
+    coverUrl = cover
 )
```
Comment (Razor0744) [inline app/src/main/java/by/zhenyabigel/bankingapplication/data/model/Account.kt]: the same I cannot get it

File: app/src/main/java/by/zhenyabigel/bankingapplication/di/DataDI.kt [hunk 8]
```diff
@@ -1,7 +1,13 @@
...
+import by.zhenyabigel.bankingapplication.data.room.AppDatabase
+import by.zhenyabigel.bankingapplication.domain.repository_interfaces.AccountsRepository
 import org.koin.dsl.module
 
 val dataModule = module {
-
+    single { AppDatabase(context = get()) }
+    single<AccountsRepository> { AccountRepositoryImpl(accountDataSource = get()) }
+    single { get<AppDatabase>().accountDao() }
+    single { get<AppDatabase>().accountDao() }
```
Comment (Razor0744) [inline app/src/main/java/by/zhenyabigel/bankingapplication/di/DataDI.kt]: space

File: app/build.gradle.kts (hunk 1 above)
Comment (zhenyabigel) [reply app/build.gradle.kts]: i did

File: app/src/main/java/by/zhenyabigel/bankingapplication/data/dao/AccountDao.kt (hunk 2 above)
Comment (zhenyabigel) [reply app/src/main/java/by/zhenyabigel/bankingapplication/data/dao/AccountDao.kt]: i did

File: app/src/main/java/by/zhenyabigel/bankingapplication/data/dao/TransactionDao.kt (hunk 3 above)
Comment (zhenyabigel) [reply app/src/main/java/by/zhenyabigel/bankingapplication/data/dao/TransactionDao.kt]: I did

File: app/src/main/java/by/zhenyabigel/bankingapplication/data/datasource/AccountDataSource.kt (hunk 4 above)
Comment (zhenyabigel) [reply app/src/main/java/by/zhenyabigel/bankingapplication/data/datasource/AccountDataSource.kt]: i did

File: app/src/main/java/by/zhenyabigel/bankingapplication/data/entities/TransactionEntity.kt (hunk 6 above)
Comment (zhenyabigel) [reply app/src/main/java/by/zhenyabigel/bankingapplication/data/entities/TransactionEntity.kt]: > don't use const like this

i did

File: app/src/main/java/by/zhenyabigel/bankingapplication/data/entities/TransactionEntity.kt (hunk 5 above)
Comment (zhenyabigel) [reply app/src/main/java/by/zhenyabigel/bankingapplication/data/entities/TransactionEntity.kt]: deleted

File: app/src/main/java/by/zhenyabigel/bankingapplication/data/model/Account.kt (hunk 7 above)
Comment (zhenyabigel) [reply app/src/main/java/by/zhenyabigel/bankingapplication/data/model/Account.kt]: i did

**Pull request merged**